src/pulse3D/plate_recording.py

```diff
@@ -374,6 +375,28 @@ def _process_plate_data(self, calibration_recordings, use_mean_of_baseline):
             well_file.displacement = np.array([adjusted_time_indices, x])
             well_file.force = calculate_force_from_displacement(well_file.displacement)
 
+    def _write_time_force_csv(self, output_dir: str):
+        # get recording name
+        if self.path.endswith(".zip") or self.path.endswith(".xlsx"):
```

Comment:
Even if there's not "." in the path, you can still use split(".")[0] and get the desired path. `os` does have a function `os.path.splitext` specifically for splitting extensions from a path. I think these 4 lines of branching can all just be `os.path.splitext(os.path.basename(self.path))[0]`